core/primitives/src/epoch_manager.rs

```diff
@@ -832,7 +832,22 @@ pub mod epoch_info {
                     shard_cps[(height as u64 % (shard_cps.len() as u64)) as usize]
                 }
                 Self::V3(v3) => {
-                    let seed = {
+                    let protocol_verstion = self.protocol_version();
+                    let seed = if checked_feature!(
+                        "stable",
+                        SynchronizeBlockChunkProduction,
+                        protocol_verstion
+                    ) && !checked_feature!(
+                        "protocol_feature_chunk_only_producers",
+                        ChunkOnlyProducers,
+                        protocol_verstion
+                    ) {
+                        // 32 bytes from epoch_seed, 8 bytes from height
```

Comment:
you should mention that here, we are using the same seed as block producer -- which should lead to matching results.